[Fix] Few prey/task hunting fixes and improvements

marcosvf132 opened the pull request:
# Description

- Prrey/task hunting free reroll timer visual issue after the player very first login only.
On player first login, there was a visual issue on the Prey/TaskHunting free reroll timer.
![image](https://user-images.githubusercontent.com/66353315/169571991-9c996368-b708-4933-aba5-ea47eb2fc332.png)

- Making prey bonus reroll more random.

- VIsual issue on prey tracker window where the bonus was not being shown when there is no monster on the list.
## Type of change

  - [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

  - [x] My code follows the style guidelines of this project
  - [x] I have performed a self-review of my own code
  - [x] I checked the PR checks reports
  - [x] My changes generate no new warnings

marcosvf132 committed:
Initial commit

marcosvf132 committed:
Merge branch 'fix-prey-time' of https://github.com/marcosvf132/canary into fix-prey-time

Codemaufeito reviewed (approved):
testado e fixou  o problema

marcosvf132 committed:
Merge branch 'opentibiabr:main' into fix-prey-time

marcosvf132 committed:
Few prey improvements

marcosvf132 committed:
Revert "Few prey improvements"

This reverts commit 6d716a5d4946385e2f2ebbdc2d7e6af5eaf43094.

marcosvf132 committed:
Fix some prey issues

- Change 'normal_random' to 'uniform_random', to make it full random.
- Change prey selection type when rerolling the slot, since the bonus is maintained.

marcosvf132 committed:
Prey 'SelectionChangeMonster' window

marcosvf132 committed:
Update ioprey.h

marcosvf132 committed:
Update ioprey.cpp

dudantas committed:
Merge branch 'main' into fix-prey-time
